libraries/wallet/wallet.cpp

```diff
@@ -2181,23 +2240,29 @@ fc::ecc::private_key wallet_api::derive_private_key(const std::string& prefix_st
                 FC_ASSERT(wit->owner == witness_account_name);
                 ap = wit->props;
             }
-#define SET_PROP(X) {p.X = !!props.X ? *(props.X) : ap.X;}
-            SET_PROP(account_creation_fee);
-            SET_PROP(maximum_block_size);
-            SET_PROP(sbd_interest_rate);
+#define SET_PROP(cp, X) {cp.X = !!props.X ? *(props.X) : ap.X;}
+            SET_PROP(p, account_creation_fee);
+            SET_PROP(p, maximum_block_size);
+            SET_PROP(p, sbd_interest_rate);
 #undef SET_PROP
-#define SET_PROP(X) {if (!!props.X) p.X = *(props.X); else if (!!ap.X) p.X = *(ap.X);}
-            SET_PROP(create_account_min_golos_fee);
-            SET_PROP(create_account_min_delegation);
-            SET_PROP(create_account_delegation_time);
-            SET_PROP(min_delegation);
-            SET_PROP(max_referral_interest_rate);
-            SET_PROP(max_referral_term_sec);
-            SET_PROP(max_referral_break_fee);
+#define SET_PROP(cp, X) {if (!!props.X) cp.X = *(props.X); else if (!!ap.X) cp.X = *(ap.X);}
+            SET_PROP(p, create_account_min_golos_fee);
+            SET_PROP(p, create_account_min_delegation);
+            SET_PROP(p, create_account_delegation_time);
+            SET_PROP(p, min_delegation);
+            op.props = p;
+            if (!!props.max_referral_interest_rate || !!props.max_referral_term_sec
```

Comment:
The logic of `wallet_api::update_chain_properties()` is focused on replace only changed values.
You break it. 
An example of broken logic:
1. User set `max_referral_interest_rate` with one call of cli_wallet
2. After some time he decide to change `account_creation_fee`
3. cli_wallet will send `chain_properties_18` which flushes `max_referral_interest_rate` back to default value.

You should check hardfork version of blockchain and use suitable version of chain_properties.